module/scheduler_components/worker.cpp

```diff
@@ -200,25 +218,9 @@ void Worker::ProcessRemoteReadResult(const internal::RemoteReadResult& read_resu
 void Worker::ExecuteAndCommitTransaction(TxnId txn_id) {
   const auto& state = txn_states_[txn_id];
   auto txn = state.txn_holder->GetTransaction();
-  auto holder = state.txn_holder;
-  switch(RemasterManager::CheckCounters(holder, storage_)) {
-    case VerifyMasterResult::VALID: {
-      // TODO: the transaction shouldn't execute until the other partitions
-      // have confirmed that the counters are valid
-      ExecuteTransactionHelper(txn_id);
-      break;
-    }
-    case VerifyMasterResult::ABORT: {
-      txn->set_status(TransactionStatus::ABORTED);
-      break;
-    }
-    case VerifyMasterResult::WAITING: {
-      LOG(ERROR) << "Transaction " << txn_id << " was sent to worker with a high counter";
-      break;
-    }
-    default:
-      LOG(ERROR) << "Unrecognized check counter result";
-      break;
+
+  if (txn->status() != TransactionStatus::ABORTED) {
+    ApplyWrites(txn_id);
```

Comment:
The name "ApplyWrites" is not quite correct. The transaction can still abort in code execution. Maybe change it to "ExecuteAndMaybeCommit" and change the "ExecuteAndCommitTransaction" function to "TriggerTransactionExecution" or something better. Naming is hard...